applications/FreeSurfaceApplication/custom_utilities/edgebased_levelset_substep.h

```diff
@@ -295,11 +274,14 @@ namespace Kratos
             // compute minimum length of the surrounding edges
             CalculateEdgeLengths(mr_model_part.Nodes());
             // set the pressure projection to the body force value
-            array_1d<double, 3> temp = mRho * mBodyForce;
             for (ModelPart::NodesContainerType::iterator inode = mr_model_part.NodesBegin();
                  inode != mr_model_part.NodesEnd();
                  inode++)
+            {
+                array_1d<double, 3> temp = mRho * inode->FastGetSolutionStepValue(BODY_FORCE);
```

Comment:
In here you're allocating an `array_1d<double,3>` for each node. To avoid this, you can declare `temp` outside the loop.

Reply:
Done